forms/upload.go

```diff
@@ -0,0 +1,75 @@
+package forms
+
+import (
+	"mime/multipart"
+
+	"go-google-scraper-challenge/helpers"
+	"go-google-scraper-challenge/models"
+
+	"github.com/beego/beego/v2/core/logs"
+	"github.com/beego/beego/v2/core/validation"
+)
+
+type UploadForm struct {
+	File       multipart.File
+	FileHeader *multipart.FileHeader
+	User       *models.User          `valid:"Required"`
+	Keywords   []string
+}
+
+// Valid adds custom validation to upload form, sets error when the validation failed.
+func (uf *UploadForm) Valid(v *validation.Validation) {
+	if uf.File == nil || uf.FileHeader == nil {
+		err := v.SetError("File", "File cannot be empty")
+		if err == nil {
+			logs.Info("Failed to set error on validation")
+		}
+	} else {
+		fileType := helpers.GetFileType(uf.FileHeader)
+		if fileType != "text/csv" {
+			err := v.SetError("File", "Incorrect file type")
+			if err == nil {
+				logs.Info("Failed to set error on validation")
+			}
+		}
+
+		keywords, err := helpers.GetFileContent(uf.File)
+		if err != nil {
+			err := v.SetError("File", "Unreadable file")
+			if err == nil {
+				logs.Info("Failed to set error on validation")
+			}
+		}
+
+		if len(keywords) > 1000 {
```

Comment:
How about handle if csv content is empty to provide some error message to the user instead of does nothing? As the requirement described

> upload file can be of any size from 1 to 1,000 keywords.

Reply:
✅  added the validation in [f592f86](https://github.com/carryall/go-google-scraper-challenge/pull/56/commits/f592f86b583d65a3226fcc6612cac88d0489eed4)